File: DuarteQuestions/DuarteQuestions/Service/UserService.cs
using DuarteQuestions.CQRS.Users.Command.ChangePassword;
using DuarteQuestions.CQRS.Users.Command.CreateUser;
using DuarteQuestions.CQRS.Users.Command.DeleteUser;
using DuarteQuestions.CQRS.Users.Command.RestoreUser;
using DuarteQuestions.CQRS.Users.Command.UpdateUser;
using DuarteQuestions.CQRS.Users.Query.GetUserById;
using DuarteQuestions.CQRS.Users.Query.GetUserList;
using DuarteQuestions.CQRS.Users.Query.LogIn;
using DuarteQuestions.CQRS.Users.ViewModel;
using DuarteQuestions.Service.Interface;
using MediatR;
using Microsoft.EntityFrameworkCore;

namespace DuarteQuestions.Service
{
    public class UserService : IUserService
    {
        private readonly IMediator _mediator;

        public UserService(IMediator mediator)
        {
            _mediator = mediator;
        }

        public async Task<bool> CreateUser(CreateUserCommand command)
        {
            try
            {
                return await _mediator.Send(command);
            }
            catch (Exception)
            {
                throw;
            }
        }

        public async Task<bool> UpdateUser(UpdateUserCommand command)
        {
            try
            {
                return await _mediator.Send(command);
            }
            catch (Exception)
            {
                throw;
            }
        }

        public async Task<bool> DeleteUser(int id)
        {
            try
            {
                return await _mediator.Send(new DeleteUserCommand()
                {
                    Id = id
                });
            }
            catch (Exception)
            {
                throw;
            }
        }

        public async Task<IEnumerable<UserViewModel>> GetUserList(GetUserListQuery query)
        {
            try
            {
                return await _mediator.Send(query);
            }
            catch (Exception)
            {
                throw;
            }
        }

        public async Task<UserViewModel> GetUserById(int id)
        {
            try
            {
                return await _mediator.Send(new GetUserByIdQuery()
                {
                    Id = id
                });
            }
            catch (Exception)
            {
                throw;
            }
        }

        public async Task<bool> LogIn(LogInCommand command)
        {
            try
            {
                return await _mediator.Send(command);
            }
            catch (Exception)
            {
                throw;
            }
        }

        public async Task<bool> RestoreUser(RestoreUserCommand command)
        {
            try
            {
                return await _mediator.Send(command);
            }
            catch (Exception)
            {
                throw;
            }
        }

        public async Task<bool> ChangePassword(ChangePasswordCommand command)
        {
            try
            {
                return await _mediator.Send(command);
            }
            catch (Exception)
            {
                throw;
            }
        }
    }
}

File: DuarteQuestionsWebPanel/src/app/services/user.service.ts
import { Injectable, inject } from '@angular/core';
import { environment } from '../environments/environment';
import { HttpClient } from '@angular/common/http';
import { Observable, share } from 'rxjs';
import { EndPoints } from '../endpoints/endpoints/endpoints';
import { ChangePasswordCommand, CreateUserCommand, GetUserListQuery, LogInCommand, RestoreUserCommand, UpdateUserCommand, UserViewModel } from '../DTOs/models/models';

@Injectable({
  providedIn: 'root'
})
export class UserService {
  private baseApiUrl: string = environment.baseApiUrl;
  private http = inject(HttpClient);

  constructor() {}

  public createUser(command: CreateUserCommand): Observable<boolean> {
    return this.http.post<boolean>(this.baseApiUrl + "/" + EndPoints.API_USER_CREATE, command).pipe(share());
  }

  public updateUser(command: UpdateUserCommand): Observable<boolean> {
    return this.http.put<boolean>(this.baseApiUrl + "/" + EndPoints.API_USER_UPDATE, command).pipe(share());
  }

  public deleteUser(id: number): Observable<boolean> {
    return this.http.delete<boolean>(this.baseApiUrl + "/" + EndPoints.API_USER_DELETE + "/" + id).pipe(share());    
  }

  public getUserList(query: GetUserListQuery): Observable<UserViewModel[]> {
    return this.http.get<UserViewModel[]>(this.baseApiUrl + "/" + EndPoints.API_USER_GET_LIST + "?Keyword=" + query.keyword + "&GetAll=" + query.getAll).pipe(share());
  }

  public getUserById(id: number): Observable<UserViewModel> {
    return this.http.get<UserViewModel>(this.baseApiUrl + "/" + EndPoints.API_USER_GET + "/" + id).pipe(share());
  }

  public login(command: LogInCommand): Observable<boolean> {
    return this.http.post<boolean>(this.baseApiUrl + "/" + EndPoints.API_USER_LOGIN, command).pipe(share());
  }

  public restoreUser(command: RestoreUserCommand): Observable<boolean> {
    return this.http.post<boolean>(this.baseApiUrl + "/" + EndPoints.API_USER_RESTORE, command).pipe(share());
  }

  public changeUserPassword(command: ChangePasswordCommand): Observable<boolean> {
    return this.http.post<boolean>(this.baseApiUrl + "/" + EndPoints.API_USER_CHANGE_PASSWORD, command).pipe(share());
  }
}